extensions-core/google-extensions/src/main/java/org/apache/druid/storage/google/GoogleStorage.java

```diff
@@ -142,28 +146,54 @@ public GoogleStorageObjectMetadata getMetadata(
     );
   }
 
-  public void delete(final String bucket, final String path) throws IOException
+
+  /**
+   * Deletes an object in a bucket on the specified path
+
+   * A false response from GCS delete API is indicative of file not found. Any other error is raised as a StorageException
+   * and should be explicitly handled.
+   Ref: <a href="https://github.com/googleapis/java-storage/blob/v2.29.1/google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java">HttpStorageRpc.java</a>
+   *
+   * @param bucket GCS bucket
+   * @param path  Object path
+   */
+  public void delete(final String bucket, final String path)
   {
     if (!storage.get().delete(bucket, path)) {
-      throw new IOE(
-          "Failed deleting google cloud storage object [bucket: %s path: %s]",
-          bucket,
-          path
-      );
+      log.debug("Google cloud storage object to be deleted not found in bucket[%s] and path[%s].", bucket, path);
     }
   }
 
   /**
    * Deletes a list of objects in a bucket
+   * A false response from GCS delete API is indicative of file not found. Any other error is raised as a StorageException
+   * and should be explicitly handled.
+   * Ref: <a href="https://github.com/googleapis/java-storage/blob/v2.29.1/google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java">HttpStorageRpc.java</a>
    *
    * @param bucket GCS bucket
    * @param paths  Iterable for absolute paths of objects to be deleted inside the bucket
    */
-  public void batchDelete(final String bucket, final Iterable<String> paths) throws IOException
+  public void batchDelete(final String bucket, final Iterable<String> paths)
   {
-    List<Boolean> statuses = storage.get().delete(Iterables.transform(paths, input -> BlobId.of(bucket, input)));
+    final List<Boolean> statuses = storage.get().delete(Iterables.transform(paths, input -> BlobId.of(bucket, input)));
     if (statuses.contains(false)) {
-      throw new IOE("Failed deleting google cloud storage object(s)");
+      if (log.isDebugEnabled()) {
+        final List<String> failedPaths = new ArrayList<>();
+        final Iterator<String> pathIterator = paths.iterator();
+        int cursor = 0;
+        while (pathIterator.hasNext()) {
+          String failedPath = pathIterator.next();
+          if (!statuses.get(cursor)) {
+            failedPaths.add(failedPath);
+          }
+          cursor++;
+        }
+        log.debug(
+            "Google cloud storage object(s) to be deleted not found in bucket[%s]. Failed paths:%s",
+            bucket,
+            failedPaths
+        );
+      }
```

Comment:
The downside of putting this whole block of code in `log.isDebugEnabled()` condition is that it won't have test coverage unless we run the tests at debug logger level; I'm not sure if there's an easy way to do it only for a subset of tests. Regardless, I'm wondering about the usefulness of this added block of code just for debug logs. If code coverage fails and if we deem this as useful info, I think we can shrink this block a wee bit, for example the log line can just be a single line:
```
log.debug("Objects not found in bucket[%s]. Failed paths:%s", bucket, failedPaths);
```

Reply:
Yes coverage is indeed a problem in this scenario and I can't think of a way to get around it. Since it is a debug log and missing files aren't that serious a concern, I propose that we remove the `isDebugEnabled()` condition and just log the bucket to avoid the expensive computation.

Reply:
The other option of-course is to ignore coverage checks for this particular block.